### PR TITLE
[RFC][UN-14538] Fix unexpected stack trace of error

### PR DESCRIPTION
**Pull request by sardyy**
***TL;DR***: ```errors.WithMessage(..)``` messing up with the stack trace

After playing a bit with ```pkg/errors``` and its methods, I realized that the stack trace of the error outputed was not the expected.

Imagine the following scenario:

```go
file x:
line 1 -> error1 := errors.New("error 1")

file y:
line 1 -> error2 := errors.Wrap(error1, "error 2")

file z:
line 1 -> error3 := errors.WithMessage(error2, "error 3")
line 2 -> Builder().Error(error3)
```

With the current implementation, if we pass the ```error3``` to our logger, since it doesn't implement the stackTracer interface, it falls into the default case, which calls the ```buildStackTrace()``` and prints the stack trace beginning in the function caller of error3 ```file z: line 2``` and this is not the expected.

The expected  behaviour is a stack trace beginning in ```file y: line 1```, because we are using the method ```Wrap``` that sets a stack trace point. If we change the ```Wrap``` to ```WithMessage``` the stack trace should begin in ```file x: line 1```.

This PR aims to solve this problem by using the ```first stackTracer error``` in the error chain if exists. If not, fallbacks to the default.

### CODE REVIEW

File: internal/logrus_test.go [hunk 1]
```diff
@@ -85,3 +85,27 @@ func TestLogrusLoggerInvalidConfig(t *testing.T) {
...
+
+func TestFirstStackTracerInErrorChainWithOneError(t *testing.T) {
+	rootError := errors.New("root error")
+
+	_, ok := rootError.(stackTracer)
+	assert.True(t, ok)
+
+	gotError := firstStackTracerInErrorChain(rootError)
+
+	assert.Equal(t, "root error", gotError.Error())
```
Comment (jmdalmeida) [inline internal/logrus_test.go]: use `assert.EqualError` here

File: internal/logrus_test.go [hunk 2]
```diff
@@ -85,3 +85,27 @@ func TestLogrusLoggerInvalidConfig(t *testing.T) {
...
+	rootError := errors.New("root error")
+	stackTracerError := errors.Wrap(rootError, "wrap root error")
+	newContextError := errors.WithMessage(stackTracerError, "new context added")
+
+	_, ok := newContextError.(stackTracer)
+	assert.False(t, ok)
+
+	gotError := firstStackTracerInErrorChain(newContextError)
+
+	assert.Equal(t, "wrap root error: root error", gotError.Error())
```
Comment (jmdalmeida) [inline internal/logrus_test.go]: You shouldn't be testing the error message but the type cast to stackTracer.
```golang
_, ok := gotError.(stackTracer)
assert.True(t, ok)
```

Comment (jmdalmeida) [reply internal/logrus_test.go]: I see your point. I think both assertions are important actually. Use `assert.EqualError` though.
```golang
assert.EqualError(t, gotError, stackTracerError.Error())

_, ok := gotError.(stackTracer)
assert.True(t, ok)
```
